site/cds_rdm/legacy/redirector.py

```diff
@@ -59,19 +55,64 @@ def legacy_files_redirect(legacy_id, filename):
     return redirect(url_path)
 
 
+def legacy_collection_redirect(collection_name):
+    """Redirection for legacy collections."""
+    cds_community_uuid = current_app.config["CDS_REDIRECTION_COLLECTIONS_MAPPING"].get(
+        collection_name, None
+    )
+    if not cds_community_uuid:
+        raise NoResultFound
+    query_params = request.args.copy()
+    query_params["q"] = query_params.pop("p", None)
+    url_path = url_for(
+        "invenio_app_rdm_communities.communities_home",
+        pid_value=cds_community_uuid,
+        **query_params,
+    )
+    return redirect(url_path)
+
+
+def legacy_search_redirect():
+    """
+    Redirection for legacy search. Transforms the legacy URL syntax into RDM URL syntax.
+
+        /legacy?cc=<legacy collection name>... -> /communities/<rdm_community_id>?...
+        /legacy?c=<legacy collection name>... -> /communities/<rdm_community_id>?...
+        /legacy?c=<legacy collection name>&p=<query>... -> /communities/<rdm_community_id>/records?q=<query>...
+    """
+    query_params = request.args.copy()
+    # Fetch current collection if it exists
+    collection_name = query_params.pop("cc", None)
+    # If not, then fetch from collection list, URLs with only single 'c' will be redirected for now
+    if not collection_name:
+        collection_name = query_params.pop("c", None)
+    if not collection_name:
+        raise not_found_error()
+    # Add logic for other redirections from search params when we get there
+    url = url_for(
+        "cds_rdm.legacy_collection_redirect",
+        collection_name=collection_name,
+        **query_params,
```

Comment:
Does all of the query params coming from legacy make sense here? For example, the `p` parameter is used to store the query string.

Reply:
I think it does, because it is possible have a mix of old and new query params in the link.